core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -397,10 +405,10 @@ protected Node(final Environment environment, Collection<Class<? extends Plugin>
                     b.bind(HttpServer.class).toProvider(Providers.of(null));
                 };
             }
-            final DiscoveryModule discoveryModule = new DiscoveryModule(this.settings, transportService, networkService,
+
+            final DiscoveryModule discoveryModule = new DiscoveryModule(this.settings, threadPool, transportService,
+                networkService, clusterService, hostsProvider -> newZenPing(settings, threadPool, transportService, hostsProvider),
```

Comment:
I considered that, but thought it could be in a followup? I don't know how large a change it will actually be, but I figured keeping the current structure wouldn't hurt anything for now.